### PR TITLE
Add documentation for the PowerBIDatasetRefresh Operator.

### PR DESCRIPTION
**Pull request by ambika-garg**
This PR intends to add the documentation like Connection types, Operators for the Power BI Dataset Refresh Operator in Apache Airflow Site.
Refer to: https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/10.5.0/_api/airflow/providers/microsoft/azure/operators/powerbi/index.html

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
static checks failing - there is some inconistency in the provider.yaml

**Comment by ambika-garg** [comment]
I've resolved the static checks. Could you please review it.

**Review by potiuk — APPROVED** [review]

File: tests/providers/__init__.py
Comment (jscheffl) [inline tests/providers/__init__.py:1]: Can you please remove the __init__.py files below tests? They seem to be created by pre-commit from a left-over folder structure and should not be merged.

File: dev/breeze/tests/test_pytest_args_for_test_types.py [hunk 1]
```diff
@@ -166,6 +166,7 @@
                 "tests/macros",
                 "tests/notifications",
                 "tests/plugins",
+                "tests/providers",
```
Comment (jscheffl) [inline dev/breeze/tests/test_pytest_args_for_test_types.py]: This seems to be an artifact from before moving providers to top level folder. Please revert this.

**Review by jscheffl — CHANGES_REQUESTED** [review]
This looks good in general but some merge conflict artefacts seems to be left-over.

Generally Good to merge but add this marker not to merge by accident.

**Review by jscheffl — APPROVED** [review]
Thanks for the cleanup! Very pretty now!

**Pull request merged**